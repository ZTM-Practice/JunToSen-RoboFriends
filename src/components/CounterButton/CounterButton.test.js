import { shallow } from "enzyme";
import React from "react";
import CounterButton from "./CounterButton";

describe('CounterButton Component', () => {
    it('renders CounterButton Component', () => {
        const mockColor = 'blue';
        expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
    });

    it('correctly increments the counter', () => {
        const mockColor = 'blue';
        const wrapper = shallow(<CounterButton color={mockColor} />);

        wrapper.find('[id="counter"]').simulate('click');
        expect(wrapper.state()).toEqual({ count: 1 });
    });
});
