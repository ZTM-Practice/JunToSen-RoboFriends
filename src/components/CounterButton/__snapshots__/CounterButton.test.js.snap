// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component renders CounterButton Component 1`] = `ShallowWrapper {}`;
